Open Judge System/Workers/OJS.Workers.ExecutionStrategies/DotNetCoreProjectTestsExecutionStrategy.cs

```diff
@@ -1,106 +1,149 @@
+                </ItemGroup>
+                <ItemGroup>
+                    {ProjectReferencesPlaceholder}
+                </ItemGroup>
+            </Project>";
+
+        private readonly string projectReferenceTemplate =
+            $@"<ProjectReference Include=""{ProjectPathPlaceholder}"" />";
+
+        public DotNetCoreProjectTestsExecutionStrategy(Func<CompilerType, string> getCompilerPathFunc)
+            : base(getCompilerPathFunc)
+        {
+        }
+
+        private string NUnitLiteConsoleAppDirectory =>
+            Path.Combine(this.WorkingDirectory, NUnitLiteConsoleAppFolderName);
+
+        private string UserProjectDirectory =>
+            Path.Combine(this.WorkingDirectory, UserSubmissionFolderName);
+
+        public override ExecutionResult Execute(ExecutionContext executionContext)
+        {
+            DirectoryHelpers.CreateDirecory(this.NUnitLiteConsoleAppDirectory);
+            DirectoryHelpers.CreateDirecory(this.UserProjectDirectory);
+
+            var result = new ExecutionResult();
+
+            var userSubmission = executionContext.FileContent;
+
+            this.ExtractFilesInWorkingDirectory(userSubmission, this.UserProjectDirectory);
+            this.ExtractTestNames(executionContext.Tests);
+
+            this.WriteTestFiles(executionContext.Tests, this.NUnitLiteConsoleAppDirectory);
+            this.WriteSetupFixture(this.NUnitLiteConsoleAppDirectory);
+
+            var userCsProjPaths = FileHelpers.FindAllFilesMatchingPattern(
+                this.UserProjectDirectory, CsProjFileSearchPattern);
+
+            var nunitLiteConsoleAppCsProjPath = this.CreateNunitLiteConsoleApp(
+                NUnitLiteConsoleAppProgramTemplate,
+                this.nunitLiteConsoleAppCsProjTemplate,
+                this.NUnitLiteConsoleAppDirectory,
+                userCsProjPaths);
+
+            var compilerPath = this.GetCompilerPathFunc(executionContext.CompilerType);
+
+            var consoleAppCompilerResult = this.Compile(
+                executionContext.CompilerType,
+                compilerPath,
+                executionContext.AdditionalCompilerArguments,
+                nunitLiteConsoleAppCsProjPath);
+
+            result.IsCompiledSuccessfully = consoleAppCompilerResult.IsCompiledSuccessfully;
+
+            if (!result.IsCompiledSuccessfully)
+            {
+                result.CompilerComment = consoleAppCompilerResult.CompilerComment;
+                return result;
+            }
+
+            // Delete tests before execution so the user can't access them
+            FileHelpers.DeleteFiles(this.TestPaths.ToArray());
+
+            var executor = new RestrictedProcessExecutor();
+            var checker = Checker.CreateChecker(
+                executionContext.CheckerAssemblyName,
+                executionContext.CheckerTypeName,
+                executionContext.CheckerParameter);
+
+            result = this.RunUnitTests(
```

Comment:
Have you tested that the output is being correctly matched by the regEx patterns from the other strategies? I.e is the output the same format as standard NUnit Console Runner output

Reply:
Yes, the output is correctly matched